omod/src/main/web/dashboardwidgets/latestobsforconceptlist/latestobsforconceptlist.controller.js

```diff
@@ -38,8 +38,33 @@ export default class LatestObsForConceptListController {
                 // Don't add obs older than maxAge
                 if (angular.isUndefined(this.maxAgeInDays) || this.widgetsCommons.dateToDaysAgo(obs.obsDatetime) <= this.maxAgeInDays) {
                     // Add last obs for concept to list
-
-                    if (['8d4a505e-c2cc-11de-8d13-0010c6dffd0f',
+                	
+                	if (angular.isDefined(obs.groupMembers) && obs.groupMembers != null) {
+                        // If obs is obs group
+                        obs.members = [];
+                        obs.value = "";
+                        angular.forEach(obs.groupMembers, (member) => {
```

Comment:
I find it confusing to have both an `obs.groupMembers` and an `obs.members` on top of/alongside it. Why not just _transform_ `obs.groupMembers` to be the array that you want?
Also it would be better practise to evaluate `config.obsGroupLabels` outside of the `forEach`. This is the kind of stuff that eventually backfires, performance wise.

Reply:
Thanks @mks-d. Yes, I'll simply transform `obs.groupMembers`. In the case of `forEach`, I would have to replicate its code three times or else creating a generic method still adds more `if, else-if` checks, so I guess we would leave the `forEach` as is.